File: micro-digiservices/ebanking/src/main/java/io/digiservices/ebanking/dto/TransactionSafDTO.java
package io.digiservices.ebanking.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class TransactionSafDTO {

    private Long numTransaction;
    private LocalDateTime dateOperation;
    private String faitPar;
    private Long numEcritureComptable;
    private BigDecimal montant;
    private BigDecimal soldeAvantOperation;
    private BigDecimal soldeApresOperation;
    private BigDecimal soldeCaisseAvant;
    private BigDecimal soldeCaisseApres;
    private String typeOperation;
    private Long typeTransaction;
    private Long sousTypeTransaction;
    private String numCompte;
    private Long serialCaisse;
    private String codeAgence;
    private String etatSaf;
    private String codeClient;
    private String motifs;
    private String codeProduit;
    private LocalDateTime dateExecution;
    private String errors;
    private Long nbre;

}

File: frontend/ultima-ng-19.0.0/src/app/interface/TransactionSafDTO.ts
export interface TransactionSafDTO {
    numTransaction: number;
    dateOperation: number[] | string; // [year, month, day, hour, minute] ou ISO string
    faitPar: string;
    numEcritureComptable: number;
    montant: number;
    soldeAvantOperation: number | null;
    soldeApresOperation: number | null;
    soldeCaisseAvant: number | null;
    soldeCaisseApres: number | null;
    typeOperation: string;
    typeTransaction: number;
    sousTypeTransaction: number;
    numCompte: string;
    serialCaisse: number;
    codeAgence: string;
    etatSaf: string;
    codeClient: string;
    motifs: string;
    codeProduit: string;
    dateExecution: number[] | string;
    errors: string | null;
    nbre: number;
}
